src/main/java/org/openrewrite/java/migrate/lang/IfElseIfConstructToSwitch.java

```diff
@@ -0,0 +1,220 @@
+            Optional<Expression> switchOn = switchOn();
+            // all ifs in the chain must be on the same variable in order to be a candidate for switch pattern matching
+            if (!switchOn.isPresent() || potentialCandidate && !patternMatchers.keySet().stream()
+                    .map(J.InstanceOf::getExpression)
+                    .allMatch(it -> SemanticallyEqual.areEqual(switchOn.get(), it))) {
+                potentialCandidate = false;
+                return;
+            }
+            boolean nullCaseInSwitch = nullCheckedParameter != null && SemanticallyEqual.areEqual(nullCheckedParameter, switchOn.get());
+            boolean hasLastElseBlock = else_ != null;
+
+            // we need at least 3 cases to use a switch
+            if (potentialCandidate && patternMatchers.keySet().size() + (nullCaseInSwitch ? 1 : 0) + (hasLastElseBlock ? 1 : 0) <= 2) {
+                potentialCandidate = false;
+            }
+        }
+
+        Optional<Expression> switchOn() {
+            return patternMatchers.keySet().stream()
+                    .map(J.InstanceOf::getExpression)
+                    .findAny();
+        }
+
+        Object[] buildTemplateArguments(Cursor cursor) {
+            Optional<Expression> switchOn = switchOn();
+            if (!switchOn.isPresent()) {
+                return new Object[0];
+            }
+            Object[] arguments = new Object[1 + (nullCheckedParameter != null ? 1 : 0) + (patternMatchers.size() * 3) + (else_ != null ? 1 : 0)];
+            arguments[0] = switchOn.get();
+            int i = 1;
+            if (nullCheckedParameter != null) {
+                // case null -> nullCheckedStatement
+                arguments[i++] = getStatementArgument(nullCheckedStatement, cursor);
+            }
+            for (Map.Entry<J.InstanceOf, Statement> entry : patternMatchers.entrySet()) {
+                J.InstanceOf instanceOf = entry.getKey();
+                // case class (pattern) -> statement
+                if (instanceOf.getClazz() instanceof J.Identifier) {
+                    arguments[i++] = ((J.Identifier) instanceOf.getClazz()).getSimpleName();
+                } else if (instanceOf.getClazz() instanceof J.FieldAccess) {
+                    arguments[i++] = ((J.FieldAccess) instanceOf.getClazz()).toString();
+                }
+                arguments[i++] = instanceOf.getPattern() == null ? "" : instanceOf.getPattern().withPrefix(Space.SINGLE_SPACE).print(cursor);
+                arguments[i++] = getStatementArgument(entry.getValue(), cursor);
+            }
+            if (else_ != null) {
+                // default -> statement
+                arguments[i] = getStatementArgument(else_, cursor);
+            }
+
+            return arguments;
+        }
+
+        String buildTemplate() {
+            StringBuilder switchBody = new StringBuilder("switch (#{any(org.openrewrite.java.tree.Expression)}) {\n");
```

Comment:
I think:
```suggestion
            StringBuilder switchBody = new StringBuilder("switch (#{any()}) {\n");
```
will probably work as well.